### PR TITLE
[docs] - Migrate Cloud agent guides [CON-340]

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR:

- Migrates the agent guides for Cloud. These were previously reviewed by the Cloud team (https://github.com/dagster-io/internal/pull/2594) and haven't changed much since that PR.
- Adds the `/dagster-cloud/deployment` category page
- Adds two new MDXComponents (`ReferenceTable`, `ReferenceTableItem`), used to create structured/HTML tables. [View an example here](https://dagster-git-erin-cloud-agent-docs-elementl.vercel.app/dagster-cloud/deployment/agents/amazon-ecs/manually-provisioning-ecs-agent#user_code_launcher-properties). (Note: The code for this in MDX files isn't pretty - I'm open to suggestions. Ideally descriptions could be in Markdown and then parsed over writing HTML.)

To do:

- [x] Address broken links
- [x] Fully implement `GenerateAgentToken` component
- [x] Address parsing in `ReferenceTableItem` component

### CODE REVIEW

File: docs/content/dagster-cloud/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc.mdx [hunk 1]
```diff
@@ -1,5 +1,85 @@
...
+  - **An existing VPC with the following:**
+    - **Public subnets**. Refer to the [AWS Work with VPCs guide](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-vpcs.html) for more info.
+    - **Enabled `enableDnsHostnames` and `enableDnsSupport` DNS attributes**. Refer to the [AWS DNS attributes documentation](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#vpc-dns-support) for more info.
+  - **Optional**: An existing ECS cluster with a [fargate capacity provider](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-capacity-providers.html). The CloudFormation template will create a cluster for you if one isn't specified.
+
+---
+
+## Step 1: Generate a Dagster Cloud agent token
+
+<!-- <GenerateAgentToken /> -->
```
Comment (johannkm) [inline docs/content/dagster-cloud/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc.mdx:37]: fancy

File: docs/next/components/mdx/includes/dagster-cloud/GenerateAgentToken.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+In this step, you'll generate a token for the agent. The Dagster Cloud agent will use this to authenticate to the agent API.
```
Comment (johannkm) [inline docs/next/components/mdx/includes/dagster-cloud/GenerateAgentToken.mdx:1]: nit, the agent vs the Dagster Cloud agent

File: docs/next/components/mdx/MDXComponents.tsx [hunk 3]
```diff
@@ -478,6 +517,9 @@ export default {
   PlaceholderImage,
   Experimental,
   Icons,
+  ReferenceTable,
+  ReferenceTableItem,
+  // GenerateAgentToken,
```
Comment (johannkm) [inline docs/next/components/mdx/MDXComponents.tsx:522]: // intentional?

File: docs/next/components/mdx/MDXComponents.tsx [hunk 4]
```diff
@@ -16,6 +16,7 @@ import "react-medium-image-zoom/dist/styles.css";
 export const SearchIndexContext = React.createContext(null);
 import path from "path";
 import { Transition } from "@headlessui/react";
+// import GenerateAgentToken from "./includes/dagster-cloud/GenerateAgentToken.mdx";
```
Comment (johannkm) [inline docs/next/components/mdx/MDXComponents.tsx:19]: nit

**Review by johannkm — COMMENTED** [review]
`ReferenceTable` stuff looks fine to me but I defer to @yuhan

File: docs/next/components/mdx/MDXComponents.tsx (hunk 3 above)
Comment (erinkcochran87) [reply docs/next/components/mdx/MDXComponents.tsx:522]: For now! I'm going to address this before merging.

File: docs/next/components/mdx/MDXComponents.tsx (hunk 4 above)
Comment (erinkcochran87) [reply docs/next/components/mdx/MDXComponents.tsx:19]: Will be addressed before merging :)

File: docs/content/dagster-cloud/deployment/agents.mdx [hunk 5]
```diff
@@ -1,7 +1,59 @@
 ---
 title: "Hybrid deployment in Dagster Cloud | Dagster Docs"
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/dagster-cloud/deployment/agents.mdx:2]: Good catch, yes it should!

**Comment by erinkcochran87** [comment]
@johannkm If this looks good to you, can you approve? We can always clean up the component later

**Review by johannkm — APPROVED** [review]

**Pull request merged**